pkg/apis/core/validation/validation.go

```diff
@@ -4232,18 +4232,12 @@ func ValidateService(service *core.Service, allowAppProtocol bool) field.ErrorLi
 
 	if service.Spec.Type == core.ServiceTypeLoadBalancer {
```

Comment:
I think this whole block can go away - the same check (`supportedPortProtocols.Has`) exists in validateServicePort() already.  Please double-check that I am not missing something